src/EditorFeatures/Core/EditAndContinue/EditAndContinueLanguageService.cs

```diff
@@ -205,7 +211,7 @@ public async ValueTask DiscardUpdatesAsync(CancellationToken cancellationToken)
                 return;
             }
 
-            _pendingUpdatedSolution = null;
+            _pendingUpdatedDesignTimeSolution = null;
```

Comment:
Do we need an Interlocked.Exchange here?

Reply:
I guess we can make this 
```C#
Contract.ThrowIfNull(Interlocked.Exchange(ref _pendingUpdatedDesignTimeSolution, null));
```

The one in CommitUpdatesAsync is just to discover bugs when debugger might call us multiple times/in parallel by mistake. Adding it here would add another guard.